consensus/polybft/oracle/oracle.go

```diff
@@ -0,0 +1,116 @@
+package oracle
+
+import (
+	"context"
+
+	"github.com/0xPolygon/polygon-edge/consensus/polybft/validator"
+	"github.com/0xPolygon/polygon-edge/types"
+	"golang.org/x/sync/errgroup"
+)
+
+// NewBlockInfo holds information about the new block
+type NewBlockInfo struct {
+	// IsEndOfEpoch indicates if the block is the last block of the epoch
+	IsEndOfEpoch bool
+	// IsFirstBlockOfEpoch indicates if the block is the first block of the epoch
+	IsFirstBlockOfEpoch bool
+	// IsEndOfSprint indicates if the block is the last block of the sprint
+	IsEndOfSprint bool
+	// FirstBlockInEpoch is the number of the first block in the epoch
+	FirstBlockInEpoch uint64
+	// CurrentEpoch is the current epoch number
+	CurrentEpoch uint64
+	// EpochSize is the number of blocks in the epoch
+	EpochSize uint64
+	// ParentBlock is the parent block of the current block
+	ParentBlock *types.Header
+	// CurrentEpochValidatorSet is the validator set for the current epoch
+	CurrentEpochValidatorSet validator.ValidatorSet
+	// NewValidatorSetDelta carries the changes in the validator set
+	NewValidatorSetDelta *validator.ValidatorSetDelta
+}
+
+func (b NewBlockInfo) CurrentBlock() uint64 {
+	return b.ParentBlock.Number + 1
+}
+
+// Oracle represents a feature that can provide and verify system transactions
+type Oracle interface {
+	// Close closes the oracle
+	Close()
+	// PostBlock posts a block to the oracle
+	PostBlock(postBlockReq *PostBlockRequest) error
+	// PostEpoch posts an epoch to the oracle
+	PostEpoch(postEpochReq *PostEpochRequest) error
```

Comment:
Just a thought:

Would it make sense to create a separate interface called e.g. `ReadOnlyOracle` or `SubscribedOracle` etc. that would contain only these two functions.

The thing is, you are currently forcing the `ProposerCalculator` to implement the `Close`, `GetTransactions` and `VerifyTransactions`, although it does not need them, so a better interface seggregation would come handy IMO (although you would need to check the interface type when you want to call a specific `Oracle`).

Reply:
Fix: https://github.com/Ethernal-Tech/blade/pull/410/commits/b48c515eef226d5465976a54583f7dd83ecfb8b9